### ISSUE TITLE
Replace link icons with Glyphicons

### ISSUE THREAD

**Issue by ahaerpfer** [dossist/moinmoin-memodump#28]
First of all a big **Thank You** for this theme!  It looks and feels fantastic and way more professional than any other available theme for MoinMoin before.

When browsing the CSS files I came across this comment in **moinizer.css**:

```
/* link type icons (would revisit later to replace with icon fonts) */
#content a[class]:before {
[...]
```

and thought I could give it a try.  I would suggest the following replacements:

| Link type | Glyphicon | Comment |
| --- | --- | --- |
| www/http/https | new-window | looks similar to the icon for external links in Mediawiki and much better than the globe, IMHO |
| file | file | obvious replacement |
| ftp | send | (paper plane) disputable, there doesn't seem to be an obvious icon for FTP |
| nntp/news | globe | disputable, any better ideas? (bullhorn? list?) |
| telnet/ssh | flash | disputable, any better ideas? |
| irc/ircs | comment | obvious replacement |
| mailto | envelope | obvious replacement |
| attachement | paperclip | obvious replacement |
| interwiki | transfer | (left/right arrows) disputable, "random" might be an alternative |
| action | cog | obvious replacement |

I already have a working patch.  If you are open to pull requests, I'd be happy to submit one.

Thoughts, further discussion, … ?

Cheers
-Andi

**Comment by ahaerpfer** [comment]
Addendum: for interwiki links the "link" glyphicon (two chain links) actually looks better than the "transfer" icon.

**Comment by dossist** [comment]
Hi, thanks for your attention and contribution!
Pull requests on unsolved issues are welcome. :)

Thanks to your detailed description, it is clear that Glyphicons lack some icons for MoinMoin, such as ftp. Though, it's cool to have bitmap icons replaced to icon fonts. I will look into the changes and discuss about it later, but it will take some time for real life reason (I think I can do it within a day!).

Just wanted to reply asap to say thanks.

**Comment by ahaerpfer** [comment]
No worries, there's real life going on at this end of the wire, too.  I've just created pull request #29 for you to review.  Take a look whenever you have time.

Cheers … and off for the weekend.
-Andi

**Comment by dossist** [comment]
Thanks for the pull request. I tested all icons and got ideas on some of link types.

| Link type | Proposal | Comment |
| --- | --- | --- |
| www/http/https | new-window | I agree to you. It would be better if icon could be attached at the end. |
| interwiki | globe + new-window? | I think this is where 'globe' icon could go. 'new-window' icon at the end to indicate that it is external. |
| ftp | floppy-disk? | no good option for this. for the time being, mocking the original bitmap icon would do, as 'send' feels like email related. (combining 'file' and 'transfer' would suit for it, but it will take some time to ensure cross browser compatibility, and css will look like an arcane magic ) |
| nntp/news | bullhorn? | 'globe' is used as a sign of an external link in some wiki clones. 'inbox' could be an alternative. |
| telnet/ssh | flash | I like your idea. 'log-in' or 'wrench' could evoke telnet/ssh, but they are confusing. |

I agree to you on other obvious ones.

Using another icon font such as Font Awesome will be another solution, but links to these protocols with no exact icons are rarely used, so I don't know if it is worth adding another font just for them.

Anyway, I will merge your code into a branch called `font-icons`, and will do tests on this topic under that branch.

Have a nice weekend!

Warmly,
dossist

**Comment by sonph** [comment]
I have a few ideas using Font Awesome:

| Link type | Icon | Comment |
| --- | --- | --- |
| www/http/https | fa-globe | or fa-external-link if you'd like to use fa-globe for interwiki |
| file | fa-file-o |  |
| ftp | fa-floppy-o | the floppy disk icon might be understood as "save" so I think fa-cloud or fa-cloud-download could be better alternatives |
| nntp/news | fa-rss or fa-bullhorn |  |
| telnet/ssh | fa-terminal | certainly the best option |
| irc/ircs | fa-comment-o |  |
| mailto | fa-envelope-o |  |
| attachment | fa-paperclip |  |
| interwiki | fa-users | based on the original MoinMoin icon |
| action | fa-cog |  |

**Comment by sonph** [comment]
Nevermind I just noticed that you are already using Glyphicons.

**Comment by dossist** [comment]
For the time being, I will go with Glyphicons.
Font Awesome is still an option worth keeping though.

**Comment by dossist** [comment]
After fine tuning, I merged the working branch to `main` in commit f77a0e262b6237f1634d01db782428cec80b6515.
Thanks to @ahaerpfer and @sonph for valuable discussion!
